src/filesystem-cache.ts

```diff
@@ -64,22 +64,70 @@ export class FilesystemCache {
   }
 
   async clearCache(key: string) {
-    fs.rmdirSync(this.getDir(key), {recursive: true});
+    if (fs.existsSync(path.join(this.getDir(key), this.cacheConfig.payloadFilename))) {
+      fs.unlink(path.join(this.getDir(key), this.cacheConfig.payloadFilename), (err) => {
+        if (err) {
+          console.error(err);
+        }
+      });
+    }
+    if (fs.existsSync(path.join(this.getDir(key), this.cacheConfig.responseFilename))) {
+      fs.unlink(path.join(this.getDir(key), this.cacheConfig.responseFilename), (err) => {
+        if (err) {
+          console.error(err);
+        }
+      });
+    }
+    if (fs.existsSync(path.join(this.getDir(key), this.cacheConfig.requestFilename))) {
+      fs.unlink(path.join(this.getDir(key), this.cacheConfig.requestFilename), (err) => {
+        if (err) {
+          console.error(err);
+        }
+      });
+    }
+    if (fs.existsSync(this.getDir(key))) {
+      fs.rmdir(this.getDir(key), () => {
+        console.log(`${key} cache Dir Deleted`);
+      });
+    }
   }
 
   async clearAllCache() {
-    fs.rmdirSync(this.getDir(''), {recursive: true});
+    fs.rmdirSync(this.getDir(''), { recursive: true });
   }
 
   cacheContent(key: string, ctx: Koa.Context) {
     const responseHeaders = ctx.response;
     const responseBody = ctx.body;
     const request = ctx.request;
-
+    // check size of stored cache to see if we are over the max number of allowed entries, and max entries isn't disabled with a value of -1 and remove over quota, removes oldest first
+    if (parseInt(this.config.cacheConfig.cacheMaxEntries) !== -1) {
+      const numCache = fs.readdirSync(this.getDir(''));
+      if (numCache.length >= parseInt(this.config.cacheConfig.cacheMaxEntries)) {
+        const toRemove = numCache.length - parseInt(this.config.cacheConfig.cacheMaxEntries) + 1;
+        let dirsDate = [];
+        for (let i = 0; i < numCache.length; i++) {
+          if (fs.existsSync(path.join(this.getDir(key), this.cacheConfig.responseFilename))) {
+            const stats = fs.statSync(path.join(this.getDir(numCache[i]), this.cacheConfig.responseFilename));
+            const mtime = stats.mtime;
+            dirsDate.push({ hash: numCache[i], age: mtime.getTime() });
+          } else {
+            dirsDate.push({ hash: numCache[i], age: 1 });
+          }
+        }
+        dirsDate.sort((a, b) => (a.age > b.age) ? 1 : -1);
```

Comment:
For simplicity, I would refactor this into a separate helper function, e.g. `sortFilesByModDate` or similar

Reply:
fixed in https://github.com/GoogleChrome/rendertron/pull/400/commits/b1a9ed8339a295187f90449eefa1323fbb171fc9